src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2648,6 +2683,10 @@ private String launchCommandIn(ArgumentListBuilder args, File workDir, EnvVars e
         if (!env.containsKey("GIT_ASKPASS")) {
             freshEnv.put("GIT_ASKPASS", "echo");
         }
+        /* Prepend extra git command line arguments if any */
+        if (!extraGitCommandArguments.isEmpty()) {
+            args = args.prepend(extraGitCommandArguments.stream().toArray(String[]::new));
```

Comment:
or more simply (and [faster](https://stackoverflow.com/a/55414113/12916)!)
```suggestion
            args = args.prepend(extraGitCommandArguments.toArray(new String[0]));
```